recipes/readerwriterqueue/all/test_package/conanfile.py

```diff
@@ -0,0 +1,14 @@
+from conans import ConanFile, CMake, tools
+import os
+
+class ConcurrentqueueTestConan(ConanFile):
+    settings = "os", "compiler", "build_type", "arch"
+    generators = "cmake"
+
+    def build(self):
+        cmake = CMake(self)
+        cmake.configure()
+        cmake.build()
+
+    def test(self):
+        self.run(os.path.join("bin", "test_package"), run_environment=True)
```

Comment:
Even though the package is header only, the host system might still be non compatible with the build system.
```suggestion
        if not tools.cross_building(self.settings):
            self.run(os.path.join("bin", "test_package"), run_environment=True)
```